File: 145.java
class Solution {
    public List<Integer> postorderTraversal(TreeNode root) {
    	List<Integer> re = new ArrayList<Integer>();
    	postorderTraversal1(root, re);
    	return re;
    }

    private void postorderTraversal1(TreeNode node,List<Integer> re) {
    	if (node == null) {
    		return;
    	}
    	postorderTraversal1(node.left, re);
    	postorderTraversal1(node.right, re);
    	re.add(node.val);
    }
}

File: 145.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def postorderTraversal(self, root: TreeNode) -> List[int]:
    	re = []
    	self.postorderTraversal1(root, re)
    	return re

    def postorderTraversal1(self, node, re):
    	if node is None:
    		return

    	self.postorderTraversal1(node.left, re)
    	self.postorderTraversal1(node.right, re)
    	re.append(node.val)